app/src/main/java/com/rockthevote/grommet/data/db/model/RockyRequest.kt

```diff
@@ -0,0 +1,211 @@
+package com.rockthevote.grommet.data.db.model
+
+import com.squareup.moshi.Json
+import com.squareup.moshi.JsonClass
+
+
+@JsonClass(generateAdapter = false)
+data class RockyRequest(
+    @Json(name = "rocky_request")
+    val rockyRequest: RockyRequestBody
+)
+
+@JsonClass(generateAdapter = false)
+data class RockyRequestBody(
+    @Json(name = "lang")
+    val lang: String,
+    @Json(name = "phone_type")
+    val phoneType: String? = null, // TODO this is specified in attributes, but not in the example Json, IDK the source
```

Comment:
>this is specified in attributes, but not in the example Json, IDK the source
you're right, it's at the top level in the attributes, but that's for canvasser info or something else, not registrant stuff, and it's nowhere in the example JSON.
![image](https://user-images.githubusercontent.com/1134161/78511805-7ff31880-776d-11ea-82d7-583b081ca03a.png)


Also in the original table definition in `RockyRequest.java`
![image](https://user-images.githubusercontent.com/1134161/78511839-b0d34d80-776d-11ea-9e0c-7154922d0de4.png)
but I think that part is a red-herring, that top level table was never meant to match the JSON structure.

So, I think this is going to be one of those Government API moments (along w/ some poor code choices, but I maintain that I was a victim of circumstance :P) :

`phone_type` isn't explicitly used in the JSON, it's a contact method.
- there are 3 contact methods: `phone`, `assistant_phone`, `email`
-- really these should not be at the same level, as phone and email are types and assistant_phone is an owner of a type

In [`AdditionalnfoFragment.java`](https://github.com/TrustTheVote-Project/Grommet/blob/master/app/src/main/java/com/rockthevote/grommet/ui/registration/AdditionalInfoFragment.java) I have Listeners on the Email and Phone, and when they are selected I update the `ContactMethod` table via listener. There is also an observable on the `PhoneType` drop down, that updates the `phone_type` column in the `rocky_request` table for the given row/registrant. 

Now the crazy part... When constructing the API call, there is some code in the  `ApiContactMethod` class that is commented out for now (had to when I removed the database stuff, but left it there for reference, it's noted in that other doc we we're tracking things in though).

When adding the contact methods for a particular registrant, if it is a `PHONE` OR `ASSISTANT_PHONE`, then I will check the `phone_type`, and if it's`mobile` I will add `SMS` as a capability.

That takes care of the `phone_type` explanation, but here's one more quirk.

It was later decided that we did not want to pass back `ASSISTANT_PHONE` as a type in the API because they realized that it could be inferred by the fact that it was a child object of the `registration_helper` object. So... no matter if it's a `PHONE` or `ASSISTANT_PHONE` we still use `PHONE` as the type. Now that I think about it, this probably means we can get rid of `ASSISTANT_PHONE` entirely it would simplify things.

Reply:
I don't know if I'm misreading or if we're talking about different fields here. The ContactMethod you're referring to was moved into this class as `ContactMethod` more or less as is, eg:
```
@JsonClass(generateAdapter = false)
data class ContactMethod(
    @Json(name = "type")
    val type: String,
    @Json(name = "value")
    val value: String,
    @Json(name = "capabilities")
    val capabilities: List<String>?
)
```
Where the logic you mention is used the same to determine `capabilities` based on `type`.

The `phone_type` in this TODO is a top level, standalone key unrelated to a ContactMethod object that doesn't have any related peers (eg, `phone_type` exists, but there isn't a `value` for the type to coincide to)